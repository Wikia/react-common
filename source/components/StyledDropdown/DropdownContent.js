import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import refPropType from '../../utils/refPropType';
/**
 * Basic DropdownContent component
 */
const DropdownContent = styled(({ elementRef, ...props }) => <div ref={elementRef} {...props} />)`
    // this needs to be empty - all the styles are defined in index
`;

DropdownContent.propTypes = {
    /**
     * React Component to display as content
     */
    children: PropTypes.node,

    /**
     * Additional class name
     */
    className: PropTypes.string,
    /**
     * React ref to the content DOM element
     */
    elementRef: refPropType.isRequired,
};

DropdownContent.defaultProps = {
    children: null,
    className: '',
};

export default DropdownContent;
