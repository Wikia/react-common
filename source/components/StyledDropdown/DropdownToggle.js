import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import IconDropdownTiny from '../../icons/IconDropdownTiny';
import IconMenuControlTiny from '../../icons/IconMenuControlTiny';

export const ChevronIcon = styled(IconMenuControlTiny)`
    color: inherit;
    fill: currentColor;
    height: 12px;
    width: 12px;
`;

function getToggleIcon(iconName) {
    return iconName === 'dropdown-tiny'
        ? <ChevronIcon as={IconDropdownTiny} />
        : <ChevronIcon />;
}

function getToggleContentComponent(toggleContent, iconName) {
    const icon = getToggleIcon(iconName);

    if (typeof toggleContent === 'function') {
        return toggleContent(icon);
    }

    if (typeof toggleContent === 'string') {
        return (
            <React.Fragment>
                <span>{toggleContent}</span>
                {icon}
            </React.Fragment>
        );
    }

    return toggleContent;
}

/**
 * Basic DropdownToggle component
 */
function DropdownToggle({
    className,
    onClick,
    attrs,
    toggleContent,
    iconName,
}) {
    const Component = attrs.href ? 'a' : 'div';

    return (
        <Component className={className} onClick={onClick} {...attrs} role="button">
            {getToggleContentComponent(toggleContent, iconName)}
        </Component>
    );
}

DropdownToggle.propTypes = {
    /**
     * HTML attributes
     */
    attrs: PropTypes.shape({
        href: PropTypes.string,
    }),
    /**
     * HTML classes
     */
    className: PropTypes.string,
    /**
     *  The icon to use for the dropdown chevron
     */
    iconName: PropTypes.oneOf([
        'dropdown-tiny',
        'menu-control-tiny',
    ]).isRequired,
    /**
     * Callback when toggle is clicked
     */
    onClick: PropTypes.func.isRequired,
    /**
     * Content of the toggle
     */
    toggleContent: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.string,
        PropTypes.node,
    ]).isRequired,
};

DropdownToggle.defaultProps = {
    className: '',
    attrs: {},
};

export default styled(DropdownToggle)`
    // this needs to be empty - all the styles are defined in index
`;
