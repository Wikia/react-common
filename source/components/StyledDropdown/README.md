Defaults:

```js
<StyledDropdown canFlip toggle="Toggle dropdown">
    <p style={{padding: '12px'}}>I am a text inside dropdown!</p>
</StyledDropdown>
```

Dropdown using dropdown-tiny chevron:
```js
<StyledDropdown canFlip toggle="Toggle dropdown" iconName="dropdown-tiny">
    <p style={{padding: '12px'}}>I am a text inside dropdown!</p>
</StyledDropdown>
```

Nested dropdown:

```js
<StyledDropdown canFlip toggle="Toggle dropdown">
    <p>I am a text inside dropdown!</p>
    <StyledDropdown toggle="Toggle dropdown" isLevel2>
        <p>I am a text inside dropdown!</p>
        <p>I am a text inside dropdown!</p>
    </StyledDropdown>
    <p>I am a text inside dropdown!</p>
</StyledDropdown>
```
