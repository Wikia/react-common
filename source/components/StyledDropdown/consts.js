
import { LIST_VERTICAL_PADDING as STYLED_LIST_VERTICAL_PADDING } from '../StyledList';

export const Z_INDEX = 1;
export const TRANSITION_DURATION = 0.2;
export const BORDER_WIDTH = 1;
export const DROPDOWN_HEIGHT = 270;
export const DROPDOWN_PADDING = 18;
export const LIST_VERTICAL_PADDING = STYLED_LIST_VERTICAL_PADDING;
