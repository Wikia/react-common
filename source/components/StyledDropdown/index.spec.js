import React from 'react';

import StyledDropdown from './index';

/* eslint-disable no-alert */

test('Dropdown renders correctly with default values', () => {
    const component = mountWithThemeProvider(
        <StyledDropdown toggle="Toggle" />,
    ).find(StyledDropdown);
    expect(component).toMatchSnapshot();
});

test('Dropdown level2 renders correctly with default values', () => {
    const component = mountWithThemeProvider(
        <StyledDropdown toggle="Toggle" isLevel2 />,
    ).find(StyledDropdown);
    expect(component).toMatchSnapshot();
});

test('Dropdown level2 renders correctly with href set', () => {
    const component = mountWithThemeProvider(
        <StyledDropdown toggle="Toggle" isLevel2 attrs={{ href: '#' }} />,
    ).find(StyledDropdown);
    expect(component).toMatchSnapshot();
});

test('Dropdown level2 renders correctly with isStickedToParent set', () => {
    const component = mountWithThemeProvider(
        <StyledDropdown toggle="Toggle" isLevel2 isStickedToParent />,
    ).find(StyledDropdown);
    expect(component).toMatchSnapshot();
});

test('Dropdown level2 renders correctly with isActive set', () => {
    const component = mountWithThemeProvider(
        <StyledDropdown toggle="Toggle" isLevel2 isActive />,
    ).find(StyledDropdown);
    expect(component).toMatchSnapshot();
});

test('Dropdown renders correctly with DropdownToggle and children', () => {
    const component = mountWithThemeProvider(
        <StyledDropdown toggle="Toggle">
            <div>Content</div>
        </StyledDropdown>,
    ).find(StyledDropdown);
    expect(component).toMatchSnapshot();
});

test('Dropdown renders correctly with DropdownToggle, toggleIconName and children', () => {
    const component = mountWithThemeProvider(
        <StyledDropdown toggle="Toggle" toggleIconName="my-icon-name">
            <div>Content</div>
        </StyledDropdown>,
    ).find(StyledDropdown);
    expect(component).toMatchSnapshot();
});

// FIXME: Re-enable once we have a legit way of testing hooks inside function
// test('Dropdown sets isClicked on click on touch device', () => {
//     const wrapper = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>,
//     ).find(StyledDropdown);

//     wrapper.setState({
//         isTouchDevice: true,
//     });

//     wrapper.find('.wds-dropdown').simulate('click', {
//         preventDefault: () => {},
//     });

//     expect(wrapper.state('isClicked')).toEqual(true);
// });

// test('Dropdown does not set isClicked on click on non-touch device', () => {
//     const wrapper = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>,
//     ).find(StyledDropdown);

//     wrapper.setState({
//         isTouchDevice: false,
//     });

//     wrapper.find('.wds-dropdown').simulate('click', {
//         preventDefault: () => {},
//     });

//     expect(wrapper.state('isClicked')).toEqual(false);
// });

// test('Dropdown sets isClicked to false on mouseLeave on touch device', () => {
//     const wrapper = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>,
//     ).find(StyledDropdown);
//     wrapper.setState({
//         isClicked: true,
//         isTouchDevice: true,
//     });

//     wrapper.find('.wds-dropdown').simulate('mouseLeave');

//     expect(wrapper.state('isClicked')).toEqual(false);
// });

// test('Dropdown does not set isClicked to false on mouseLeave on touch device', () => {
//     const wrapper = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>,
//     ).find(StyledDropdown);
//     wrapper.setState({
//         isClicked: true,
//         isTouchDevice: false,
//     });

//     wrapper.find('.wds-dropdown').simulate('mouseLeave');

//     expect(wrapper.state('isClicked')).toEqual(true);
// });

// test('onToggleClicked runs handleClick with an event and a bool flagset to true', () => {
//     const handleClickMock = sinon.spy();
//     const eventMock = { preventDefault: () => {}, stopPropagation: () => {} };
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.instance().handleClick = handleClickMock;
//     component.instance().onToggleClicked(eventMock);

//     expect(handleClickMock.calledWith(true, eventMock)).toBe(true);
// });

// test('onClick runs handleClick with a bool flag set to false', () => {
//     const handleClickMock = sinon.spy();
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.instance().handleClick = handleClickMock;
//     component.instance().onClick();

//     expect(handleClickMock.calledWith(false)).toBe(true);
// });

// test('handleClick changes isClicked state on touch device', () => {
//     const eventMock = { preventDefault: () => {}, stopPropagation: () => {} };
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.setState({ isTouchDevice: true });

//     component.instance().handleClick(true, eventMock);

//     expect(component.state('isClicked')).toBe(true);
// });

// test('handleClick does not change isClicked state on non-touch device', () => {
//     const eventMock = { preventDefault: () => {}, stopPropagation: () => {} };
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.setState({ isTouchDevice: false });

//     component.instance().handleClick(true, eventMock);

//     expect(component.state('isClicked')).toBe(false);
// });

// test('handleClick calls event methods when passed true as an argument', () => {
//     const eventMock = { preventDefault: sinon.spy(), stopPropagation: sinon.spy() };
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.setState({ isTouchDevice: true });

//     component.instance().handleClick(true, eventMock);

//     expect(eventMock.preventDefault.called).toBe(true);
//     expect(eventMock.stopPropagation.called).toBe(true);
// });

// test('handleClick calls onClose from props if dropdown is closing on mobile', () => {
//     const eventMock = { preventDefault: () => {}, stopPropagation: () => {} };
//     const onCloseMock = sinon.spy();
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle" onClose={onCloseMock}>
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.setState({ isTouchDevice: true, isClicked: true });

//     component.instance().handleClick(true, eventMock);

//     expect(onCloseMock.called).toBe(true);
// });

// test('handleClick calls default onClose if dropdown is closing on mobile', () => {
//     const eventMock = { preventDefault: () => {}, stopPropagation: () => {} };
//     const onCloseMock = sinon.spy();

//     StyledDropdown.defaultProps.onClose = onCloseMock;
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.setState({ isTouchDevice: true, isClicked: true });

//     component.instance().handleClick(true, eventMock);

//     expect(onCloseMock.called).toBe(true);
// });

// test('handleClick does not call onClose from props if dropdown is closing on desktop', () => {
//     const eventMock = { preventDefault: () => {}, stopPropagation: () => {} };
//     const onCloseMock = sinon.spy();
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle" onClose={onCloseMock}>
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.setState({ isTouchDevice: false, isClicked: true });

//     component.instance().handleClick(true, eventMock);

//     expect(onCloseMock.called).toBe(false);
// });

// test('Dropdown does not have class `wds-is-flipped` applied when canFlip is not', () => {
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle">
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.instance().contentElementRef.current.getBoundingClientRect = () => ({
//         bottom: 800,
//     });

//     component.simulate('mouseEnter');

//     expect(component.find('.wds-dropdown').hasClass('wds-is-flipped')).toBe(false);
// });

// test('Dropdown calls onMouseEnter prop on respective event', () => {
//     const onMouseEnterMock = jest.fn();
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle" onMouseEnter={onMouseEnterMock}>
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.instance().contentElementRef.current.getBoundingClientRect = () => ({
//         bottom: 800,
//     });

//     component.simulate('mouseEnter');

//     expect(onMouseEnterMock).toBeCalled();
// });


// test('Dropdown has class `wds-is-flipped` applied when canFlip is set and it is near the bottom of the viewport', () => {
//     const component = mountWithThemeProvider(
//         <StyledDropdown toggle="Toggle" canFlip>
//             <div>Content</div>
//         </StyledDropdown>
//     ).find(StyledDropdown);

//     component.instance().contentElementRef.current.getBoundingClientRect = () => ({
//         bottom: 800,
//     });

//     component.simulate('mouseEnter');

//     expect(component.find('.wds-dropdown').hasClass('wds-is-flipped')).toBe(true);

//     component.simulate('mouseLeave');

//     expect(component.find('.wds-dropdown').hasClass('wds-is-flipped')).toBe(false);
// });
