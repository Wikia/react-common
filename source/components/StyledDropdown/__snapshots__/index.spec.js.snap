// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown level2 renders correctly with default values 1`] = `
<StyledDropdown
  canFlip={false}
  className=""
  contentClassName=""
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={false}
  hasDarkShadow={false}
  hasShadow={false}
  iconName="menu-control-tiny"
  isActive={false}
  isLevel2={true}
  isNotHoverable={false}
  isStickedToParent={false}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle="Toggle"
  toggleAttrs={Object {}}
  toggleClassName=""
>
  <StyledDropdown__DropdownComponent
    as="li"
    className="is-level-2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <li
      className="StyledDropdown__DropdownComponent-sc-1w57ze8-0 bVobta is-level-2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <DropdownToggle
        attrs={Object {}}
        className=""
        iconName="menu-control-tiny"
        isLevel2={true}
        isTouchDevice={false}
        onClick={[Function]}
        toggleContent="Toggle"
      >
        <DropdownToggle
          attrs={Object {}}
          className="DropdownToggle-sc-1p907bm-1 gvIzNe"
          iconName="menu-control-tiny"
          isLevel2={true}
          isTouchDevice={false}
          onClick={[Function]}
          toggleContent="Toggle"
        >
          <div
            className="DropdownToggle-sc-1p907bm-1 gvIzNe"
            onClick={[Function]}
            role="button"
          >
            <span>
              Toggle
            </span>
            <DropdownToggle__ChevronIcon>
              <_default
                className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
              >
                <_default
                  className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                >
                  <div
                    className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                  />
                </_default>
              </_default>
            </DropdownToggle__ChevronIcon>
          </div>
        </DropdownToggle>
      </DropdownToggle>
      <DropdownContent
        className=""
        elementRef={
          Object {
            "current": <div
              class="DropdownContent-sc-1m1tfah-0 dKeZyS"
            />,
          }
        }
      >
        <Component
          className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          elementRef={
            Object {
              "current": <div
                class="DropdownContent-sc-1m1tfah-0 dKeZyS"
              />,
            }
          }
        >
          <div
            className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          />
        </Component>
      </DropdownContent>
    </li>
  </StyledDropdown__DropdownComponent>
</StyledDropdown>
`;

exports[`Dropdown level2 renders correctly with href set 1`] = `
<StyledDropdown
  attrs={
    Object {
      "href": "#",
    }
  }
  canFlip={false}
  className=""
  contentClassName=""
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={false}
  hasDarkShadow={false}
  hasShadow={false}
  iconName="menu-control-tiny"
  isActive={false}
  isLevel2={true}
  isNotHoverable={false}
  isStickedToParent={false}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle="Toggle"
  toggleAttrs={Object {}}
  toggleClassName=""
>
  <StyledDropdown__DropdownComponent
    as="li"
    className="is-level-2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <li
      className="StyledDropdown__DropdownComponent-sc-1w57ze8-0 bVobta is-level-2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <DropdownToggle
        attrs={Object {}}
        className=""
        iconName="menu-control-tiny"
        isLevel2={true}
        isTouchDevice={false}
        onClick={[Function]}
        toggleContent="Toggle"
      >
        <DropdownToggle
          attrs={Object {}}
          className="DropdownToggle-sc-1p907bm-1 gvIzNe"
          iconName="menu-control-tiny"
          isLevel2={true}
          isTouchDevice={false}
          onClick={[Function]}
          toggleContent="Toggle"
        >
          <div
            className="DropdownToggle-sc-1p907bm-1 gvIzNe"
            onClick={[Function]}
            role="button"
          >
            <span>
              Toggle
            </span>
            <DropdownToggle__ChevronIcon>
              <_default
                className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
              >
                <_default
                  className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                >
                  <div
                    className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                  />
                </_default>
              </_default>
            </DropdownToggle__ChevronIcon>
          </div>
        </DropdownToggle>
      </DropdownToggle>
      <DropdownContent
        className=""
        elementRef={
          Object {
            "current": <div
              class="DropdownContent-sc-1m1tfah-0 dKeZyS"
            />,
          }
        }
      >
        <Component
          className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          elementRef={
            Object {
              "current": <div
                class="DropdownContent-sc-1m1tfah-0 dKeZyS"
              />,
            }
          }
        >
          <div
            className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          />
        </Component>
      </DropdownContent>
    </li>
  </StyledDropdown__DropdownComponent>
</StyledDropdown>
`;

exports[`Dropdown level2 renders correctly with isActive set 1`] = `
<StyledDropdown
  canFlip={false}
  className=""
  contentClassName=""
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={false}
  hasDarkShadow={false}
  hasShadow={false}
  iconName="menu-control-tiny"
  isActive={true}
  isLevel2={true}
  isNotHoverable={false}
  isStickedToParent={false}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle="Toggle"
  toggleAttrs={Object {}}
  toggleClassName=""
>
  <StyledDropdown__DropdownComponent
    as="li"
    className="is-active is-level-2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <li
      className="StyledDropdown__DropdownComponent-sc-1w57ze8-0 bVobta is-active is-level-2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <DropdownToggle
        attrs={Object {}}
        className=""
        iconName="menu-control-tiny"
        isLevel2={true}
        isTouchDevice={false}
        onClick={[Function]}
        toggleContent="Toggle"
      >
        <DropdownToggle
          attrs={Object {}}
          className="DropdownToggle-sc-1p907bm-1 gvIzNe"
          iconName="menu-control-tiny"
          isLevel2={true}
          isTouchDevice={false}
          onClick={[Function]}
          toggleContent="Toggle"
        >
          <div
            className="DropdownToggle-sc-1p907bm-1 gvIzNe"
            onClick={[Function]}
            role="button"
          >
            <span>
              Toggle
            </span>
            <DropdownToggle__ChevronIcon>
              <_default
                className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
              >
                <_default
                  className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                >
                  <div
                    className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                  />
                </_default>
              </_default>
            </DropdownToggle__ChevronIcon>
          </div>
        </DropdownToggle>
      </DropdownToggle>
      <DropdownContent
        className=""
        elementRef={
          Object {
            "current": <div
              class="DropdownContent-sc-1m1tfah-0 dKeZyS"
            />,
          }
        }
      >
        <Component
          className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          elementRef={
            Object {
              "current": <div
                class="DropdownContent-sc-1m1tfah-0 dKeZyS"
              />,
            }
          }
        >
          <div
            className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          />
        </Component>
      </DropdownContent>
    </li>
  </StyledDropdown__DropdownComponent>
</StyledDropdown>
`;

exports[`Dropdown level2 renders correctly with isStickedToParent set 1`] = `
<StyledDropdown
  canFlip={false}
  className=""
  contentClassName=""
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={false}
  hasDarkShadow={false}
  hasShadow={false}
  iconName="menu-control-tiny"
  isActive={false}
  isLevel2={true}
  isNotHoverable={false}
  isStickedToParent={true}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle="Toggle"
  toggleAttrs={Object {}}
  toggleClassName=""
>
  <StyledDropdown__DropdownComponent
    as="li"
    className="is-level-2 is-sticked-to-parent"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <li
      className="StyledDropdown__DropdownComponent-sc-1w57ze8-0 bVobta is-level-2 is-sticked-to-parent"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <DropdownToggle
        attrs={Object {}}
        className=""
        iconName="menu-control-tiny"
        isLevel2={true}
        isTouchDevice={false}
        onClick={[Function]}
        toggleContent="Toggle"
      >
        <DropdownToggle
          attrs={Object {}}
          className="DropdownToggle-sc-1p907bm-1 gvIzNe"
          iconName="menu-control-tiny"
          isLevel2={true}
          isTouchDevice={false}
          onClick={[Function]}
          toggleContent="Toggle"
        >
          <div
            className="DropdownToggle-sc-1p907bm-1 gvIzNe"
            onClick={[Function]}
            role="button"
          >
            <span>
              Toggle
            </span>
            <DropdownToggle__ChevronIcon>
              <_default
                className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
              >
                <_default
                  className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                >
                  <div
                    className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                  />
                </_default>
              </_default>
            </DropdownToggle__ChevronIcon>
          </div>
        </DropdownToggle>
      </DropdownToggle>
      <DropdownContent
        className=""
        elementRef={
          Object {
            "current": <div
              class="DropdownContent-sc-1m1tfah-0 dKeZyS"
            />,
          }
        }
      >
        <Component
          className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          elementRef={
            Object {
              "current": <div
                class="DropdownContent-sc-1m1tfah-0 dKeZyS"
              />,
            }
          }
        >
          <div
            className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          />
        </Component>
      </DropdownContent>
    </li>
  </StyledDropdown__DropdownComponent>
</StyledDropdown>
`;

exports[`Dropdown renders correctly with DropdownToggle and children 1`] = `
<StyledDropdown
  canFlip={false}
  className=""
  contentClassName=""
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={false}
  hasDarkShadow={false}
  hasShadow={false}
  iconName="menu-control-tiny"
  isActive={false}
  isLevel2={false}
  isNotHoverable={false}
  isStickedToParent={false}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle="Toggle"
  toggleAttrs={Object {}}
  toggleClassName=""
>
  <StyledDropdown__DropdownComponent
    as="div"
    className=""
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="StyledDropdown__DropdownComponent-sc-1w57ze8-0 bVobta"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <DropdownToggle
        attrs={Object {}}
        className=""
        iconName="menu-control-tiny"
        isLevel2={false}
        isTouchDevice={false}
        onClick={[Function]}
        toggleContent="Toggle"
      >
        <DropdownToggle
          attrs={Object {}}
          className="DropdownToggle-sc-1p907bm-1 gvIzNe"
          iconName="menu-control-tiny"
          isLevel2={false}
          isTouchDevice={false}
          onClick={[Function]}
          toggleContent="Toggle"
        >
          <div
            className="DropdownToggle-sc-1p907bm-1 gvIzNe"
            onClick={[Function]}
            role="button"
          >
            <span>
              Toggle
            </span>
            <DropdownToggle__ChevronIcon>
              <_default
                className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
              >
                <_default
                  className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                >
                  <div
                    className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                  />
                </_default>
              </_default>
            </DropdownToggle__ChevronIcon>
          </div>
        </DropdownToggle>
      </DropdownToggle>
      <DropdownContent
        className=""
        elementRef={
          Object {
            "current": <div
              class="DropdownContent-sc-1m1tfah-0 dKeZyS"
            >
              <div>
                Content
              </div>
            </div>,
          }
        }
      >
        <Component
          className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          elementRef={
            Object {
              "current": <div
                class="DropdownContent-sc-1m1tfah-0 dKeZyS"
              >
                <div>
                  Content
                </div>
              </div>,
            }
          }
        >
          <div
            className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          >
            <div>
              Content
            </div>
          </div>
        </Component>
      </DropdownContent>
    </div>
  </StyledDropdown__DropdownComponent>
</StyledDropdown>
`;

exports[`Dropdown renders correctly with DropdownToggle, toggleIconName and children 1`] = `
<StyledDropdown
  canFlip={false}
  className=""
  contentClassName=""
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={false}
  hasDarkShadow={false}
  hasShadow={false}
  iconName="menu-control-tiny"
  isActive={false}
  isLevel2={false}
  isNotHoverable={false}
  isStickedToParent={false}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle="Toggle"
  toggleAttrs={Object {}}
  toggleClassName=""
  toggleIconName="my-icon-name"
>
  <StyledDropdown__DropdownComponent
    as="div"
    className=""
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="StyledDropdown__DropdownComponent-sc-1w57ze8-0 bVobta"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <DropdownToggle
        attrs={Object {}}
        className=""
        iconName="menu-control-tiny"
        isLevel2={false}
        isTouchDevice={false}
        onClick={[Function]}
        toggleContent="Toggle"
      >
        <DropdownToggle
          attrs={Object {}}
          className="DropdownToggle-sc-1p907bm-1 gvIzNe"
          iconName="menu-control-tiny"
          isLevel2={false}
          isTouchDevice={false}
          onClick={[Function]}
          toggleContent="Toggle"
        >
          <div
            className="DropdownToggle-sc-1p907bm-1 gvIzNe"
            onClick={[Function]}
            role="button"
          >
            <span>
              Toggle
            </span>
            <DropdownToggle__ChevronIcon>
              <_default
                className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
              >
                <_default
                  className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                >
                  <div
                    className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                  />
                </_default>
              </_default>
            </DropdownToggle__ChevronIcon>
          </div>
        </DropdownToggle>
      </DropdownToggle>
      <DropdownContent
        className=""
        elementRef={
          Object {
            "current": <div
              class="DropdownContent-sc-1m1tfah-0 dKeZyS"
            >
              <div>
                Content
              </div>
            </div>,
          }
        }
      >
        <Component
          className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          elementRef={
            Object {
              "current": <div
                class="DropdownContent-sc-1m1tfah-0 dKeZyS"
              >
                <div>
                  Content
                </div>
              </div>,
            }
          }
        >
          <div
            className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          >
            <div>
              Content
            </div>
          </div>
        </Component>
      </DropdownContent>
    </div>
  </StyledDropdown__DropdownComponent>
</StyledDropdown>
`;

exports[`Dropdown renders correctly with default values 1`] = `
<StyledDropdown
  canFlip={false}
  className=""
  contentClassName=""
  contentScrollable={false}
  dropdownLeftAligned={false}
  dropdownRightAligned={false}
  hasDarkShadow={false}
  hasShadow={false}
  iconName="menu-control-tiny"
  isActive={false}
  isLevel2={false}
  isNotHoverable={false}
  isStickedToParent={false}
  noChevron={false}
  onClose={null}
  onMouseEnter={null}
  toggle="Toggle"
  toggleAttrs={Object {}}
  toggleClassName=""
>
  <StyledDropdown__DropdownComponent
    as="div"
    className=""
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="StyledDropdown__DropdownComponent-sc-1w57ze8-0 bVobta"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <DropdownToggle
        attrs={Object {}}
        className=""
        iconName="menu-control-tiny"
        isLevel2={false}
        isTouchDevice={false}
        onClick={[Function]}
        toggleContent="Toggle"
      >
        <DropdownToggle
          attrs={Object {}}
          className="DropdownToggle-sc-1p907bm-1 gvIzNe"
          iconName="menu-control-tiny"
          isLevel2={false}
          isTouchDevice={false}
          onClick={[Function]}
          toggleContent="Toggle"
        >
          <div
            className="DropdownToggle-sc-1p907bm-1 gvIzNe"
            onClick={[Function]}
            role="button"
          >
            <span>
              Toggle
            </span>
            <DropdownToggle__ChevronIcon>
              <_default
                className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
              >
                <_default
                  className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                >
                  <div
                    className="DropdownToggle__ChevronIcon-sc-1p907bm-0 IqZgt"
                  />
                </_default>
              </_default>
            </DropdownToggle__ChevronIcon>
          </div>
        </DropdownToggle>
      </DropdownToggle>
      <DropdownContent
        className=""
        elementRef={
          Object {
            "current": <div
              class="DropdownContent-sc-1m1tfah-0 dKeZyS"
            />,
          }
        }
      >
        <Component
          className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          elementRef={
            Object {
              "current": <div
                class="DropdownContent-sc-1m1tfah-0 dKeZyS"
              />,
            }
          }
        >
          <div
            className="DropdownContent-sc-1m1tfah-0 dKeZyS"
          />
        </Component>
      </DropdownContent>
    </div>
  </StyledDropdown__DropdownComponent>
</StyledDropdown>
`;
